src/test/scala/is/hail/nativecode/NativeCodeSuite.scala

```diff
@@ -50,4 +50,82 @@ class NativeCodeSuite extends SparkSuite {
     assert(a.use_count() == 1)
   }
 
+  @Test def testNativeGlobal() = {
+    var ret: Long = -1
+    val st = new NativeStatus()
+    val globalModule = new NativeModule("global")
+    val funcHash1 = globalModule.findLongFuncL1(st, "hailTestHash1")
+    if (st.fail) error(s"${st}")
+    assert(st.ok)
+    val funcHash8 = globalModule.findLongFuncL8(st, "hailTestHash8")
+    if (st.fail) error(s"${st}")
+    assert(st.ok)
+    ret = funcHash8(st, 0x1, 0x2, 0x3, 0x4, 0x5, 0x6, 0x7, 0x8)
+    assert(ret == 0x87654321L)
+    val t0 = System.currentTimeMillis()
+    var sum: Long = 0
+    val numCalls = 100*1000000
+    var countdown = numCalls
+    while (countdown > 0) {
+      sum = funcHash1(st, sum)
+      countdown -= 1
+    }
+    val t1 = System.currentTimeMillis()
+    val usecsPerCall = ((t1 - t0) * 1000.0) / numCalls
+    info(s"funcHash1() ~ ${usecsPerCall}usecs")
+    assert(usecsPerCall < 0.2)
+  }
+
+  @Test def testNativeBuild() = {
+    val sb = new StringBuilder()
+    sb.append("#include \"hail/hail.h\"\n")
```

Comment:
Scala multi-line strings might make this a bit more managable:

```
scala> """This
     | is a multi-line
     | string!"""
res2: String =
This
is a multi-line
string!
```

Reply:
Yes, that's neater